pkg/text/truncate.go

```diff
@@ -27,49 +22,15 @@ func Truncate(maxWidth int, s string) string {
 		return s
 	}
 
-	useEllipsis := false
+	tail := ""
 	if maxWidth >= minWidthForEllipsis {
-		useEllipsis = true
-		maxWidth -= ellipsisWidth
+		tail = ellipsis
 	}
 
-	g := uniseg.NewGraphemes(s)
-	r := ""
-	rWidth := 0
-	for {
-		g.Next()
-		gWidth := graphemeWidth(g)
-
-		if rWidth+gWidth <= maxWidth {
-			r += g.Str()
-			rWidth += gWidth
-			continue
-		} else {
-			break
-		}
-	}
-
-	if useEllipsis {
-		r += "..."
-	}
-
-	if rWidth < maxWidth {
+	r := truncate.StringWithTail(s, uint(maxWidth), tail)
```

Comment:
I'm wondering is the new truncate implementation compatible with our old one. We iterated over "graphemes" to avoid ever slicing one in half due to truncation. Does `github.com/muesli/reflow/truncate` do that too?

Reply:
Yes, `github.com/muesli/reflow/truncate` uses `github.com/mattn/go-runewidth`. I made sure before I switched. Not only does it simplify the code to maintain here, but also counts and emits the tail as needed automatically.